### PR TITLE
Graph tracking model architecture changes

### PR DESCRIPTION
**Pull request by R-Ding**
## What
* Create keras layer `Unmerge` used to unmerge embeddings and centroids.
* Change default `merge_type` for `get_embedding_temporal_merge_model` and `get_delta_temporal_merge_model` to lstm instead of cnn
* Remove residual connections in `get_tracking_decoder` 

## Why
* Nicer to have a keras layer than use Lambda layers
* Merging the embeddings depends only on timepoints in the present and past, and thus do not have information from future embeddings. Cnn's use information from future timepoints, resulting in temporal information being leaked. Lstm's only use information from the current and old embeddings (which is what we want).
* Adding the residual connections does not help with accuracy

### CODE REVIEW

File: deepcell/model_zoo/tracking.py [hunk 1]
```diff
@@ -351,7 +364,7 @@ def get_appearance_encoder(self):
         x = TimeDistributed(ImageNormalization2D(norm_method='whole_image',
                                                  name='imgnrm_ae'))(x)
 
-        for i in range(self.time_window):
+        for i in range(5):
```
Comment (willgraf) [inline deepcell/model_zoo/tracking.py:367]: Any reason to hardcode 5 rather than leave it as a configurable parameter?

File: deepcell/model_zoo/tracking.py [hunk 2]
```diff
@@ -287,7 +310,7 @@ def __init__(self,
         # Create model
         self.training_model, self.inference_model = self.get_models()
 
-    def get_embedding_temporal_merge_model(self, merge_type='cnn'):
+    def get_embedding_temporal_merge_model(self, merge_type='lstm'):
```
Comment (willgraf) [inline deepcell/model_zoo/tracking.py:313]: If only LSTM works, let's just remove the "cnn" option and `merge_type` parameter entirely.

Comment (willgraf) [reply deepcell/model_zoo/tracking.py:313]: Same with `get_delta_temporal_merge_model`.

**Review by willgraf — CHANGES_REQUESTED** [review]

File: deepcell/model_zoo/tracking.py (hunk 1 above)
Comment (R-Ding) [reply deepcell/model_zoo/tracking.py:367]: Dave’s reasoning is we only want to do 5 convolution and pooling operations and it’s alright to have it hardcoded for now as the crop size (32) is also currently hardcoded.

**Review by willgraf — APPROVED** [review]
I'm a little uncomfortable with hardcoding the the conv/pool ops as well as hardcoding the appearance shape. But the appearance shape is hardcoded in the main branch so we can resolve that before merging the PR.

I'd like @MekWarrior to review and approve as well, but this looks like a good improvement.

File: deepcell/model_zoo/tracking.py [hunk 3]
```diff
@@ -233,12 +233,35 @@ def call(self, inputs):
         return output
 
 
+class Unmerge(tf.keras.layers.Layer):
```
Comment (MekWarrior) [inline deepcell/model_zoo/tracking.py:236]: Should this be moved to the `layers` module?

**Review by MekWarrior — APPROVED** [review]
I agree with Will that it strikes me as odd to hardcode something when the variable is already there. But If it works, it's better than our current version.

File: deepcell/model_zoo/tracking.py (hunk 3 above)
Comment (R-Ding) [reply deepcell/model_zoo/tracking.py:236]: I was wondering that too, there's also two other layers defined here, `Comparison` and `DeltaReshape`. Should we move all three into an existing `layers` file or a new one?

Comment (willgraf) [reply deepcell/model_zoo/tracking.py:236]: We should probably move them into a new layers file. IDK what to name it, probably should be more about the temporal/graph nature rather than tracking.

Each of them should have tests as well, but they cannot use the standard testing routine since they have multiple inputs.

I think the `FilterDetections` layer has some good tests that are not using the `layer_test` method.

Comment (MekWarrior) [reply deepcell/model_zoo/tracking.py:236]: We have  `temporal`set up already - that may be a good fit.

Comment (willgraf) [reply deepcell/model_zoo/tracking.py:236]: I'm going to merge this PR and update the TODO checklist to include migrating and testing the layers.
